crm_claim_rma/models/claim_line.py

```diff
@@ -0,0 +1,413 @@
+        return True
+
+    @api.returns('stock.location')
+    def get_destination_location(self, product_id, warehouse_id):
+        """
+        Compute and return the destination location to take
+        for a return. Always take 'Supplier' one when return type different
+        from company.
+        """
+        location_dest_id = warehouse_id.lot_stock_id
+
+        if product_id.seller_ids:
+            seller = product_id.seller_ids[0]
+            if seller.warranty_return_partner != 'company' \
+                    and seller.name and \
+                    seller.name.property_stock_supplier:
+                location_dest_id = seller.name.property_stock_supplier
+
+        return location_dest_id
+
+    def _warranty_return_address_values(self, product, company, warehouse):
+        """
+        Return the partner to be used as return destination and
+        the destination stock location of the line in case of return.
+
+        We can have various cases here:
+            - company or other: return to company partner or
+              crm_return_address_id if specified
+            - supplier: return to the supplier address
+        """
+        if not (product and company and warehouse):
+            return {
+                'warranty_return_partner': False,
+                'warranty_type': False,
+                'location_dest_id': False
+            }
+        sellers = product.seller_ids
+        if sellers:
+            seller = sellers[0]
+            return_address_id = seller.warranty_return_address.id
+            return_type = seller.warranty_return_partner
+        else:
+            # when no supplier is configured, returns to the company
+            return_address = (company.crm_return_address_id or
+                              company.partner_id)
+            return_address_id = return_address.id
+            return_type = 'company'
+        location_dest = self.get_destination_location(product, warehouse)
+        return {
+            'warranty_return_partner': return_address_id,
+            'warranty_type': return_type,
+            'location_dest_id': location_dest.id
+        }
+
+    def set_warranty_return_address(self):
+        self.ensure_one()
+        claim = self.claim_id
+        values = self._warranty_return_address_values(
+            self.product_id, claim.company_id, claim.warehouse_id)
+        self.write(values)
+        return True
+
+    @api.multi
+    def set_warranty(self):
+        """
+        Calculate warranty limit and address
+        """
+        for line_id in self:
+            if not line_id.product_id:
+                raise exceptions.UserError(_('Please set product first'))
+
+            if not line_id.invoice_line_id:
+                raise exceptions.UserError(_('Please set invoice first'))
+
+            line_id.set_warranty_limit()
+            line_id.set_warranty_return_address()
+
+    @api.model
+    def _get_sequence_number(self):
+        """
```

Comment:
First line empty in method docstring, according to C0199(docstring-first-line-empty)